var/spack/repos/builtin/packages/eckit/package.py

```diff
@@ -145,13 +149,6 @@ def cmake_args(self):
             # args.append('-DBUILD_SHARED_LIBS=OFF')
             raise InstrallError("eckit static build not supported")
 
-        if '+mpi' in self.spec:
-            args += [
-                '-DCMAKE_C_COMPILER=%s' % self.spec['mpi'].mpicc,
-                '-DCMAKE_CXX_COMPILER=%s' % self.spec['mpi'].mpicxx,
-                '-DCMAKE_Fortran_COMPILER=%s' % self.spec['mpi'].mpifc,
-            ]
```

Comment:
What's the problem with setting the compiler to be the MPI wrapper?

Reply:
One problem is that if you compile and link everything with the MPI wrappers, you have serial executables as part of the package that now have the MPI dependencies. On systems that are configured to use slurm AND force you to go through libpmi/libslurm for any parallel execution, this means you can't run a serial executable without srun anymore. Depending on how the system is configured, you may even have to submit it to a compute node.

There was also some discussion in the spack channels and issues about the disadvantages of linking everything with the MPI wrappers.

Reply:
Ok, I see. MPI is optional. This is the same reason in hpc-stack we build a serial/parallel version of certain packages like NetCDF and Nemsio.